Helper/Webhook/WebhookHandlerFactory.php

```diff
@@ -12,252 +12,94 @@
-        self::$adyenLogger = $adyenLogger;
-        self::$chargedCurrency = $chargedCurrency;
-        self::$configHelper = $configHelper;
-        self::$invoiceHelper = $invoiceHelper;
-        self::$adyenOrderPaymentFactory = $adyenOrderPaymentFactory;
-        self::$magentoInvoiceFactory = $magentoInvoiceFactory;
-        self::$paymentMethodsHelper = $paymentMethodsHelper;
-        self::$adyenOrderPaymentCollectionFactory = $adyenOrderPaymentCollectionFactory;
-        self::$vaultHelper = $vaultHelper;
-        self::$paymentRequest = $paymentRequest;
-        self::$agreementCollectionFactory = $agreementCollectionFactory;
-        self::$agreementFactory = $agreementFactory;
-        self::$billingAgreementResourceModel = $billingAgreementResourceModel;
-        self::$tokenManagement = $paymentTokenManagement;
-        self::$paymentTokenFactory = $paymentTokenFactory;
-        self::$encryptor = $encryptor;
-        self::$paymentTokenRepository = $paymentTokenRepository;
-        self::$orderPaymentResourceModel = $orderPaymentResourceModel;
+        $this->adyenLogger = $adyenLogger;
+        $this->authorisationWebhookHandler = $authorisationWebhookHandler;
+        $this->captureWebhookHandler = $captureWebhookHandler;
+        $this->offerClosedWebhookHandler = $offerClosedWebhookHandler;
+        $this->refundWebhookHandler = $refundWebhookHandler;
+        $this->refundFailedWebhookHandler = $refundFailedWebhookHandler;
+        $this->manualReviewAcceptWebhookHandler = $manualReviewAcceptWebhookHandler;
+        $this->manualReviewRejectWebhookHandler = $manualReviewRejectWebhookHandler;
+        $this->recurringContractWebhookHandler = $recurringContractWebhookHandler;
+        $this->pendingWebhookHandler = $pendingWebhookHandler;
+        $this->cancellationWebhookHandler = $cancellationWebhookHandler;
+        $this->cancelOrRefundWebhookHandler = $cancelOrRefundWebhookHandler;
+        $this->orderClosedWebhookHandler = $orderClosedWebhookHandler;
     }
 
     /**
-     * @param string $eventCode
-     * @return WebhookHandlerInterface
      * @throws InvalidDataException
      */
-    public static function create(string $eventCode): WebhookHandlerInterface
+    public function create(string $eventCode): WebhookHandlerInterface
     {
         switch ($eventCode) {
             case Notification::HANDLED_EXTERNALLY:
             case Notification::AUTHORISATION:
-                return new AuthorisationWebhookHandler(
-                    self::$adyenOrderPayment,
-                    self::$orderHelper,
-                    self::$caseManagementHelper,
-                    self::$serializer,
-                    self::$adyenLogger,
-                    self::$chargedCurrency,
-                    self::$configHelper,
-                    self::$invoiceHelper,
-                    self::$paymentMethodsHelper
-                );
+                return $this->authorisationWebhookHandler;
```

Comment:
I think there is a way to remove this switch statement to be in the config and DI but this is Okay also